File: 03-1-login_kt2/app/src/test/java/com/example/login_kt2/repository/UserRepositoryTest.kt
package com.example.login_kt2.repository

import kotlinx.coroutines.runBlocking
import org.junit.Test
import org.junit.Assert.*

class UserRepositoryTest {

    val userRepository = UserRepository(true)

    @Test
    fun loginSuccessful() = runBlocking {
        val res = userRepository.login("user1", "123")
        assertEquals("success", res)
    }

    @Test
    fun loginWrongUsername() = runBlocking {
        val res = userRepository.login("andre", "123")
        assertEquals("wrong_username", res)
    }

    @Test
    fun loginWrongPassword() = runBlocking {
        val res = userRepository.login("user1", "123wrong")
        assertEquals("wrong_password", res)
    }
}

File: 03-1-login_fl/test/repository/user_repository_test.dart
import 'package:test/test.dart';
import 'package:login_fl/repository/user_repository.dart';

// run tests, >> dart test -r expanded

void main() {
  final userRepository = UserRepository(true);

  test('login successful', () async {
    final response = await userRepository.login('user1', '123');
    expect(response, equals('success'));
  });

  test('login wrong user', () async {
    final response = await userRepository.login('user5', '123');
    expect(response, equals('wrong_username'));
  });

  test('login wrong password', () async {
    final response = await userRepository.login('user1', '456');
    expect(response, equals('wrong_password'));
  });
}